desc/compute/utils.py

```diff
@@ -691,15 +690,22 @@ def surface_integrals(grid, q=jnp.array([1]), surface_label="rho", max_surface=F
     # 'nodes', so that the duplicate surface has nodes with the same
     # surface label and is treated as one, like in the previous paragraph.
 
-    integrals = jnp.histogram(nodes_modulo, bins=bins, weights=ds * q)[0]
+    def integrate_component(integrands):
```

Comment:
I think `vmap` could get compiled each time `surface_integrals` is called because `integrate_component` is a nested function. Would this alternative avoid that
```python
def surface_integrals(...):
    ....
    vector_of_integrals = jnp.atleast_1d(
        jnp.squeeze(vmap_integrate_component(nodes_modulo, bins, vector_of_integrands))
    )

def vmap_integrate_component(nodes_modulo, bins, vector_of_integrands):
    return vmap(functools.partial(integrate_component, nodes_modulo, bins), out_axes=1)(vector_of_integrands)

def integrate_component(nodes_modulo, bins, integrands):
    return jnp.histogram(nodes_modulo, bins=bins, weights=integrands)[0]
```

Reply:
I don't think we have to worry too much. For most of the regular stuff the compilation doesn't add much overhead (mainly on derivatives). JAX is also usually pretty good about dealing with inlined functions